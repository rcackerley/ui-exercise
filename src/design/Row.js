// @flow strict @format

import React from "react"
import type { Node } from "react"

import injectSheet from "react-jss"

import classNames from "classnames"

type Props = {|
  classes: { [string]: string },
  customStyles: string,
  children: Node,
  centered: boolean,
  onClick: ?() => void
|}

const styles = {
  row: {
    display: "flex"
  },
  center: {
    alignItems: "center"
  }
}

const Row = ({ classes, children, customStyles, centered, onClick }: Props) => (
  <div
    onClick={onClick}
    className={classNames(
      classes.row,
      customStyles,
      centered && classes.center
    )}
  >
    {children}
  </div>
)

export default injectSheet(styles)(Row)
