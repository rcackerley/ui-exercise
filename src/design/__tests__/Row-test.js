import React from "react"
import ReactDOM from "react-dom"
import Row from "../Row"

import renderer from "react-test-renderer"

it("render Row component correctly", () => {
  const RowComponent = renderer.create(<Row />).toJSON()
  expect(RowComponent).toMatchSnapshot()
})
