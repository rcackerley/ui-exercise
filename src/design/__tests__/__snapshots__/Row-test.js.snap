// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Row component correctly 1`] = `
<div
  className="Row-row-0-1-1"
/>
`;
